uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js

```diff
@@ -57,8 +84,38 @@ module.exports = (function () {
       return this;
     },
 
+    /**
+     * Selects the request Absence Type by the given label
+     *
+     * @param  {String} absenceTypeLabel ex. "Holiday in Hours"
+     * @return {Promise}
+     */
+    selectRequestAbsenceType: function (absenceTypeLabel) {
+      var absenceTypeSelect, absenceTypeOptionIndex;
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.evaluate(function(absenceTypeLabel) {
+          absenceTypeSelect = document.querySelector('[name=absenceTypeSelect]');
```

Comment:
you could use jQuery here too:

```js
casper.evaluate(function (absenceTypeLabel) {
  var absenceTypeSelect = jQuery('[name=absenceTypeSelect]');
  var absenceTypeOption = absenceTypeSelect.find('option:contains(' + absenceTypeLabel + ')');

  // .change() triggers the change event for the select:
  absenceTypeSelect.val(absenceTypeOption.attr('value')).change();
, absenceTypeLabel);
```

Reply:
I am not sure we want to have that dependency. I know jQuery is everywhere, but I prefer native code in this case.

Reply:
Rewrote as @AkA84 suggested